std/math.d

```diff
@@ -4103,9 +4103,55 @@ long lrint(real x) @trusted pure nothrow @nogc
 
             return sign ? -result : result;
         }
+        else static if (F.realFormat == RealFormat.ieeeQuadruple)
+        {
+            long result;
+
+            // Rounding limit when casting from real(128-bit) to ulong.
+            enum real OF = 0x1P+112;
+
+            ulong* vl = cast(ulong*)(&x);
+
+            enum EXPSHIFT_ULONG = 48;
+            enum MANTISSE = real.mant_dig - 1;
+
+            // Find the exponent and sign
+            ulong hi = vl[MANTISSA_MSB];
+            int exp = cast(int)((hi >> EXPSHIFT_ULONG) & F.EXPMASK) - F.EXPBIAS;
+            int sign = cast(int)(hi >> 63);
+
+            if (exp < 63)
+            {
+                // Adjust x and check result.
+                real j = sign ? -OF : OF;
+                x = (j + x) - j;
+                hi = vl[MANTISSA_MSB];
+                exp = cast(int)((hi >> EXPSHIFT_ULONG) & F.EXPMASK) - F.EXPBIAS;
+                hi &= 0xffff_ffffffff;
+                hi |= 0x10000_00000000;
+
+                if (exp < 0)
+                    result = 0;
+                else if (exp <= 48)
+                    result = cast(long) hi >> (EXPSHIFT_ULONG - exp);
+                else
+                {
+                    const lo = vl[MANTISSA_LSB];
+                    result = (cast(long) hi << (exp - EXPSHIFT_ULONG)) | (lo >> (MANTISSE - exp));
+                }
+            }
+            else
+            {
+                // It is left implementation defined when the number is too large
+                // to fit in a 64bit long.
+                return cast(long) x;
```

Comment:
@wilzbach 
> so this function should return `cent` then?

No - it should return `long` for now, to be compatible with algorithms that depend on the existing hard-coded return type.

Long-term, all these overly-similar rounding functions - `round()`, `rint()`, `lrint()`, and `nearbyint()` - could be replaced by a single template function. One of the template parameters can be the return type, allowing `round!long()` to directly replace `lrint()`. `round!cent()` would be supported as well.